File: src/api/metadata.ts
import { invoke } from '@tauri-apps/api/core';
import type { MetadataInput, MetadataOutput } from '../types/metadata';

export async function getAudioMetadata(filePath: string): Promise<MetadataOutput> {
    return await invoke<MetadataOutput>('get_audio_metadata', { filePath });
}

export async function setAudioMetadata(input: MetadataInput): Promise<void> {
    await invoke('set_audio_metadata', { input });
}

File: src-tauri/src/commands/metadata.rs
use crate::services::{read_metadata, write_metadata};
use crate::types::{MetadataInput, MetadataOutput};

#[tauri::command]
pub async fn get_audio_metadata(file_path: String) -> Result<MetadataOutput, String> {
    read_metadata(&file_path)
}

#[tauri::command]
pub async fn set_audio_metadata(input: MetadataInput) -> Result<(), String> {
    write_metadata(&input)
}
